x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_enrichment/processor_status_indicator.tsx

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiAvatar,
+  EuiAvatarProps,
+  EuiLoadingSpinner,
+  EuiToolTip,
+  useEuiTheme,
+} from '@elastic/eui';
+import { useSelector } from '@xstate5/react';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { ProcessorConfigurationProps } from './processors';
+import {
+  useSimulatorSelector,
+  useStreamEnrichmentSelector,
+} from './state_management/stream_enrichment_state_machine';
+import { selectWhetherAnyProcessorBeforePersisted } from './state_management/stream_enrichment_state_machine/selectors';
+
+export const ProcessorStatusIndicator = ({
+  processorRef,
+}: {
+  processorRef: ProcessorConfigurationProps['processorRef'];
+}) => {
+  const { euiTheme } = useEuiTheme();
+
+  const isNew = useSelector(processorRef, (snapshot) => snapshot.context.isNew);
+
+  const hasSimulation = useSimulatorSelector((snapshot) => Boolean(snapshot.context.simulation));
+  const isParticipatingInSimulation = useSimulatorSelector((snapshot) =>
+    snapshot.context.processors.find((p) => p.id === processorRef.id)
+  );
+  const isSimulationRunning = useSimulatorSelector((snapshot) =>
+    snapshot.matches('runningSimulation')
+  );
+  const isPending = useSimulatorSelector(
+    (snapshot) =>
+      snapshot.context.simulation &&
+      !snapshot.context.simulation.processors_metrics[processorRef.id]
+  );
+  const isFailing = useSimulatorSelector((snapshot) =>
+    Boolean(
+      snapshot.context.simulation?.processors_metrics[processorRef.id]?.errors.some(
+        (e) => e.type === 'generic_simulation_failure'
+      )
+    )
+  );
+  const isAnyProcessorBeforePersisted = useStreamEnrichmentSelector(
+    selectWhetherAnyProcessorBeforePersisted
+  );
+
+  let variant:
+    | (EuiAvatarProps & {
+        tooltipContent?: string;
+      })
+    | null = null;
+
+  if (isAnyProcessorBeforePersisted) {
+    const name = i18n.translate(
+      'xpack.streams.streamDetailView.managementTab.enrichment.mixedProcessorTooltip',
+      {
+        defaultMessage:
+          'Currently, is not possible to simulate persisted and new processors together. Move the new processors after the persisted processors to restore the simulation behavior.',
```

Comment:
```suggestion
          'Simulation fails when new processors are added before existing processors. To restore the simulation, move new processors after existing processors.',
```

Not sure if I'm understanding the "persisting processors" language. If what i've suggested doesn't make sense, I might need a little more explanation.

Reply:
Sure, basically when a user adds a new processor, it's always added to the end of the list, in the right place to run the simulation. The user also has the chance to reorder new processors (non-saved yet) with previously created once. In that case, we cannot reliably run the simulation an we disable it for all the processors.

Reply:
I updated the suggestion. Let me know if that works for you.